test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDisplayExtensionsTest.cs

```diff
@@ -355,6 +414,22 @@ public void DisplayForModel_UsesTemplateNameAndHtmlFieldName()
         private class SomeModel
         {
             public string SomeProperty { get; set; }
+
+            public Status Status { get; set; }
+        }
+
+        private class StatusResource
+        {
+            public static string Type_Fail = "failure from type";
+        }
+
+        private enum Status : byte
+        {
+            [Display(Name = "Resx_Pass")]
+            Pass,
+            [Display(Name = "Type_Fail", ResourceType = typeof(StatusResource))]
```

Comment:
This is in no way a failure. Just a different way to override the display name.

Need to cover `[Display(ResourceType = typeof(StatusResource))]` here.

Separately, please confirm the four cases for these files (after the above addition) are covered when the string localizer factory is `null`.